etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2476 @@
+* @see [naming-convention]{@link https://typescript-eslint.io/rules/naming-convention}
+*
+* @example
+* // Bad...
+* const BAD_NAME = 1;
+* const bad_Name = 2;
+*
+* // Good...
+* const goodName = 3;
+* const GOOD_NAME = 4;
+*/
+rules[ '@typescript-eslint/naming-convention' ] = [
+	'error',
+	{
+		'selector': 'variable',
+		'format': [ 'camelCase', 'UPPER_CASE' ],
+		'leadingUnderscore': 'forbid',
+		'trailingUnderscore': 'forbid'
+	}
+];
+
+/**
+* Disables TypeScript's no-shadow rule.
+*
+* @name no-shadow
+* @memberof rules
+* @type {Array}
+* @default [ 'off', { 'hoist': 'all' }]
+* @see [no-shadow]{@link https://typescript-eslint.io/rules/no-shadow}
+*
+* @example
+* // Okay...
+* const x = 1;
+* function example() {
+*   const x = 2;
+* }
+*/
+rules[ '@typescript-eslint/no-shadow' ] = [ 'off', {
+	'hoist': 'all'
+}];
+
+/**
+* Disables the rule against the delete operator with computed key expressions.
+*
+* @name no-dynamic-delete
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-dynamic-delete]{@link https://typescript-eslint.io/rules/no-dynamic-delete}
+*
+* @example
+* // Okay...
+* delete object[dynamicKey];
+*/
+rules[ '@typescript-eslint/no-dynamic-delete' ] = 'off';
+
+/**
+* Disallows empty functions. Functions with a comment inside are not considered empty.
+*
+* @name no-empty-function
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-empty-function]{@link https://typescript-eslint.io/rules/no-empty-function}
+*
+* @example
+* // Bad...
+* function noOp() {}
+*
+* // Good...
+* function noOp() {
+*   // intentional no-op
+* }
+*/
+rules[ '@typescript-eslint/no-empty-function' ] = 'error';
+
+/**
+* Disallows the declaration of empty interfaces.
+*
+* @name no-empty-interface
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-empty-interface]{@link https://typescript-eslint.io/rules/no-empty-interface}
+*
+* @example
+* // Bad...
+* interface Empty {}
+*
+* // Good...
+* interface NotEmpty {
+*   data: string;
+* }
+*/
+rules[ '@typescript-eslint/no-empty-interface' ] = 'error';
+
+/**
+* Warns on usage of the `any` type.
+*
+* @name no-explicit-any
+* @memberof rules
+* @type {string}
+* @default 'warn'
+* @see [no-explicit-any]{@link https://typescript-eslint.io/rules/no-explicit-any}
+*
+* @example
+* // Bad...
+* function unsafe(arg: any): any { return arg; }
+*
+* // Good...
+* function safe(arg: unknown): unknown { return arg; }
+*/
+rules[ '@typescript-eslint/no-explicit-any' ] = 'warn';
+
+/**
+* Disallows classes that only have static members and don't need to be instantiated.
+*
+* @name no-extraneous-class
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-extraneous-class]{@link https://typescript-eslint.io/rules/no-extraneous-class}
+*
+* @example
+* // Bad...
+* class StaticOnly {
+*   static doSomething() {}
+* }
+*
+* // Good...
+* namespace StaticOnly {
+*   export function doSomething() {}
```

Comment:
```suggestion
*     export function doSomething() {}
```

Reply:
@Planeshifter Would you mind ensuring that indentation conventions are consistent with what we use for JSDoc in JavaScript files? Namely, 4 spaces. That should be applied here and elsewhere.